File: 118_pascalTriangle.cpp
vector<vector<int>> generate(int numRows)
{
	vector<vector<int>> res;
	if(numRows <= 0)
		return res;
	vector<int> last(1, 1);
	res.push_back(last);
	for(int i=1; i<numRows; i++)
	{
		last.push_back(0);
		vector<int> path = last;
		for(int j=1; j<=last.size()-1; j++)
			path[j] = last[j] + last[j-1];
		res.push_back(path);
		last = path;
	}
	return res;
}

File: Algorithms/118_PascalTriangle.java
public List<List<Integer>> generate(int numRows) {
	List<List<Integer>> res = new ArrayList<List<Integer>>();
	List<Integer> tmp = new ArrayList<Integer>();
	for(int i=1; i<=numRows; i++) {
		tmp.add(0, 1);
		for(int j=1; j<=tmp.size()-2; j++)
			tmp.set(j, tmp.get(j)+tmp.get(j+1));
		res.add(new ArrayList<Integer>(tmp));
	}
	return res;
}